src/clients/python/simple_callback_client.py

```diff
@@ -0,0 +1,157 @@
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+from functools import partial
+import numpy as np
+import os
+import threading
+from builtins import range
+from tensorrtserver.api import *
+
+FLAGS = None
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--verbose', action="store_true", required=False, default=False,
+                        help='Enable verbose output')
+    parser.add_argument('-u', '--url', type=str, required=False, default='localhost:8000',
+                        help='Inference server URL. Default is localhost:8000.')
+    parser.add_argument('-i', '--protocol', type=str, required=False, default='http',
+                        help='Protocol ("http"/"grpc") used to ' +
+                        'communicate with inference service. Default is "http".')
+
+    FLAGS = parser.parse_args()
+    protocol = ProtocolType.from_str(FLAGS.protocol)
+
+    # We use a simple model that takes 2 input tensors of 16 integers
+    # each and returns 2 output tensors of 16 integers each. One
+    # output tensor is the element-wise sum of the inputs and one
+    # output is the element-wise difference.
+    model_name = "simple"
+    model_version = -1
+    batch_size = 1
+
+    # Create the inference context for the model.
+    ctx = InferContext(FLAGS.url, protocol, model_name, model_version, FLAGS.verbose)
+
+    # Create the data for the two input tensors. Initialize the first
+    # to unique integers and the second to all ones.
+    input0_data = np.arange(start=0, stop=16, dtype=np.int32)
+    input1_data = np.ones(shape=16, dtype=np.int32)
+
+    # Send inference request to the inference server. Get results for
+    # both output tensors.
+    class Context:
+        def __init__(self, input0_data, input1_data):
+            self._lock = threading.Lock()
+            self._cv = threading.Condition(self._lock)
+            self._input0_data = input0_data
+            self._input1_data = input1_data
+            self._request_cnt = 2
+            self._done_cnt = 0
+            self._recent_ctx_obj = None
+            self._recent_request_id = None
+            self._success = True
+    
+    context = Context(input0_data, input1_data)
+    def active_callback(context, idx, ctx_obj, request_id):
```

Comment:
Why is it called active_callback instead of completion_callback. 'ctx_obj' is also generic, how about 'infer_ctx'. This example is also confusing because there is a lot of stuff in that user data and it doesn't seem to be best practice. Can you instead just grab infer_ctx and request_id and schedule a future or enqueue or something that then performs the checking and error reporting.